src/Selenium2Driver.php

```diff
@@ -425,7 +427,45 @@ public function back()
 
     public function switchToWindow(?string $name = null)
     {
-        $this->getWebDriverSession()->focusWindow($name ?: '');
+        if ($name === null) {
+            $windowHandles = $this->getWebDriverSession()->window_handles();
+            $name = self::W3C_WINDOW_HANDLE_PREFIX .  reset($windowHandles);
+        }
+
+        if (is_string($name)) {
+            $name = $this->getWindowHandleFromName($name);
+        }
+
+        $this->getWebDriverSession()->focusWindow($name);
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function getWindowHandleFromName(string $name): string
+    {
+        // if name is actually prefixed window handle, just remove the prefix
+        if (strpos($name, self::W3C_WINDOW_HANDLE_PREFIX) === 0) {
+            return substr($name, strlen(self::W3C_WINDOW_HANDLE_PREFIX));
+        }
+
+        // ..otherwise check if any existing window has the specified name
+
+        $origWindowHandle = $this->getWebDriverSession()->window_handle();
+
+        try {
+            foreach ($this->getWebDriverSession()->window_handles() as $handle) {
```

Comment:
should be deduplicated with self::getWindowNames()

Reply:
The `self::getWindowNames` method iterates all windows and returns an array with window names only.

@mvorisek , How it can help in this method, that needs to figure out what window handle matches to a given window name?

P.S.
This method does fewer iterations (=selenium API calls) in the case when the searched window isn't the last opened.

Reply:
I prefer to completely remove this hack - https://github.com/minkphp/MinkSelenium2Driver/pull/384#discussion_r1501840465.

Reply:
Resolving.

All further discussion about using window handles instead of names should happen in the https://github.com/minkphp/Mink/issues/868 .